internal/kafka/avail_status_msg.go

```diff
@@ -19,15 +23,27 @@ func NewAvailabilityStatusMessage(msg *GenericMessage) (*AvailabilityStatusMessa
 	return &asm, nil
 }
 
-func (m AvailabilityStatusMessage) GenericMessage() (GenericMessage, error) {
+func (m AvailabilityStatusMessage) GenericMessage(ctx context.Context) (GenericMessage, error) {
+	return genericMessage(ctx, m, m.SourceID, AvailabilityStatusRequestTopic)
+}
+
+func genericMessage(ctx context.Context, m any, key string, topic string) (GenericMessage, error) {
 	payload, err := json.Marshal(m)
 	if err != nil {
 		return GenericMessage{}, fmt.Errorf("unable to marshal message: %w", err)
 	}
 
+	id := ctxval.Identity(ctx)
+
 	return GenericMessage{
-		Topic: AvailabilityStatusRequestTopic,
-		Key:   []byte(m.SourceID),
+		Topic: topic,
+		Key:   []byte(key),
 		Value: payload,
+		Headers: map[string]string{
+			"Content-Type":                "application/json",
+			"x-rh-identity":               identity.GetIdentityHeader(ctx),
+			"x-rh-sources-org-id":         id.Identity.OrgID,
+			"X-RH-Sources-Account-Number": id.Identity.AccountNumber,
+		},
```

Comment:
Okay  this is good, but God I hate mixed style. Can you sync up on either all lowercase or "Camel-Case-Style"? :-D